src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -142,7 +142,9 @@ internal void SetBuffer(byte[] buffer, int offset, int count)
 
         internal bool TryGetBuffer(out ArraySegment<byte> buffer)
         {
-            return _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            bool result = _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            Debug.Assert(result, "Performance: TryGetBuffer failed, which means that reads from the content will copy the data!");
```

Comment:
@stephentoub 

> We already assert at call sites if we expected to be able to get the buffer and can't

That does not seem to be the case: see [here](https://github.com/jamesqo/corefx/blob/784a167a96ce69e245eed808c0e42c3401b13101/src/System.Net.Http/src/System/Net/Http/HttpContent.cs#L187), [here](https://github.com/jamesqo/corefx/blob/784a167a96ce69e245eed808c0e42c3401b13101/src/System.Net.Http/src/System/Net/Http/HttpContent.cs#L280), and [here](https://github.com/jamesqo/corefx/blob/784a167a96ce69e245eed808c0e42c3401b13101/src/System.Net.Http/src/System/Net/Http/HttpContent.cs#L244).